docs/csharp/programming-guide/classes-and-structs/object-and-collection-initializers.md

```diff
@@ -54,25 +54,25 @@ var pet = new { Age = 10, Name = "Fluffy" };
   
  When this query is executed, the `productInfos` variable will contain a sequence of objects that can be accessed in a `foreach` statement as shown in this example:  
   
-```  
+```csharp
 foreach(var p in productInfos){...}  
 ```  
   
  Each object in the new anonymous type has two public properties which receive the same names as the properties or fields in the original object. You can also rename a field when you are creating an anonymous type; the following example renames the `UnitPrice` field to `Price`.  
   
-```  
+```csharp
 select new {p.ProductName, Price = p.UnitPrice};  
 ```  
   
 ## Object initializers with nullable types  
  It is a compile-time error to use an object initializer with a nullable struct.  
   
 ## Collection initializers  
- Collection initializers let you specify one or more element initializers when you initialize a collection class that implements <xref:System.Collections.IEnumerable> or a class with an `Add` extension method. The element initializers can be a simple value, an expression or an object initializer. By using a collection initializer you do not have to specify multiple calls to the `Add` method of the class in your source code; the compiler adds the calls.  
+ Collection initializers let you specify one or more element initializers when you initialize a collection type that implements <xref:System.Collections.IEnumerable> and has `Add` with appropriate signature as instance method or extension method. The element initializers can be a simple value, an expression or an object initializer. By using a collection initializer you do not have to specify multiple calls to the `Add` method of the class in your source code; the compiler adds the calls.  
```

Comment:
with appropriate signature --> with the appropriate signature
as instance method or extension method --> as an instance method or an extension method